rtt-target/src/init.rs

```diff
@@ -137,6 +138,20 @@ macro_rules! rtt_init {
         #[export_name = "_SEGGER_RTT"]
         pub static mut CONTROL_BLOCK: MaybeUninit<RttControlBlock> = MaybeUninit::uninit();
 
+        #[allow(unused)]
+        #[export_name = "rtt_init_must_not_be_called_multiple_times"]
+        fn rtt_init_must_not_be_called_multiple_times() { }
+
+        use ::rtt_target::export::critical_section;
+
+        static INITIALIZED: critical_section::Mutex<Cell<bool>> = critical_section::Mutex::new(Cell::new(false));
+        critical_section::with(|cs| {
+            if INITIALIZED.borrow(cs).get() {
```

Comment:
Why do we need the runtime check if we have a compiletime check already?

Reply:
because it doesn't help against placing the macro in a function, then calling the function twice.

Reply:
Ohh, good point!